### PR TITLE
Resolve lifetime issues related to macro dependencies (part 2)

### PR DESCRIPTION
**Pull request by slavek-kucera**

### CODE REVIEW

File: parser_library/src/lsp/item_convertors.h [hunk 1]
```diff
@@ -45,6 +48,18 @@ completion_item_s generate_completion_item(const context::sequence_symbol& sym);
...
+    const std::function<std::vector<std::string>(std::string_view)>& instruction_suggestions);
+std::vector<completion_item_s> generate_completion(
+    const std::unordered_map<context::id_index, std::unique_ptr<context::sequence_symbol>>*,
+    const std::function<std::vector<std::string>(std::string_view)>& instruction_suggestions);
+std::vector<completion_item_s> generate_completion(const completion_list_instructions&,
+    const std::function<std::vector<std::string>(std::string_view)>& instruction_suggestions);
+
 } // namespace hlasm_plugin::parser_library::lsp
 
 #endif
```
Comment (jirimosinger) [inline parser_library/src/lsp/item_convertors.h:65]: Consider adding a new line
```suggestion

} // namespace hlasm_plugin::parser_library::lsp

#endif

```

File: parser_library/test/gtest_stringers.cpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                  << "\n range: " << item.occurence_range << " }";
+}
+
+std::ostream& operator<<(std::ostream& stream, const lsp::completion_item_s& item)
+{
+    return stream << "{ label: " << item.label << "\n detail: " << item.detail << "\n insert text: " << item.insert_text
+                  << "\n documentation: " << item.documentation << "\n kind: " << (int)item.kind << " }";
+}
+
+} // namespace hlasm_plugin::parser_library::lsp
```
Comment (jirimosinger) [inline parser_library/test/gtest_stringers.cpp:70]: Consider adding a new line

```suggestion
} // namespace hlasm_plugin::parser_library::lsp

```

File: parser_library/test/lsp/lsp_completion.cpp [hunk 3]
```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2019 Broadcom.
```
Comment (jirimosinger) [inline parser_library/test/lsp/lsp_completion.cpp:2]:
```suggestion
 * Copyright (c) 2023 Broadcom.
```

File: parser_library/src/lsp/item_convertors.cpp [hunk 4]
```diff
@@ -225,10 +226,115 @@ completion_item_s generate_completion_item(const macro_info& sym, const file_inf
...
+        [&instruction_suggestions](auto v) { return generate_completion(v, instruction_suggestions); }, cls);
+}
+
+completion_list_s generate_completion(std::monostate, const std::function<std::vector<std::string>(std::string_view)>&)
+{
+    return completion_list_s();
+}
+
+completion_list_s generate_completion(
+    const vardef_storage* var_defs, const std::function<std::vector<std::string>(std::string_view)>&)
```
Comment (jirimosinger) [inline parser_library/src/lsp/item_convertors.cpp:249]: Can `var_defs` be `null` for any reason?

**Review by jirimosinger — APPROVED** [review]

File: parser_library/src/lsp/item_convertors.cpp (hunk 4 above)
Comment (slavek-kucera) [reply parser_library/src/lsp/item_convertors.cpp:249]: not really, these functions work as visitors, so they need to match the type.

**Pull request merged**